core/cap-0034.md

```diff
@@ -0,0 +1,352 @@
+we combine, in the sense that today, if changing any given parameter `A` from
+`1` to `2` and changing any given parameter `B` from `1` to `2` would each be
+sensible individually, then it would also be sensible to make both changes at
+once. But if the type of possible upgrades could expand to one in which certain
+combinations of parameter changes would be insensible, then it might no longer
+make sense just to combine each parameter individually -- the simple union of
+individual changes could produce a result which was inconsistent. In that case,
+the right way to combine them consistently would become ambiguous; there could
+be multiple options of which none would include all candidate parameter changes.
+That would not be inherently wrong -- we have the same situation with
+transaction sets, and we make a heuristic but deterministic choice as to which
+is the best. But it might be some argument for choosing a single nominated
+upgrade in the same way as we already choose a single nominated tx set, and are
+proposing to choose a single nominated closetime, just because choosing the best
+single nominated set of upgrades could be simpler (there might be far fewer
+options to choose among) than choosing the best consistent combination of all
+tx sets.
+
+- The selection of a single `StellarValue` from the set of candidates
+during the ballot protocol reflects the literature on and formal models of
+the Stellar Consensus Protocol better than the current implementation's
+combination of candidates.
+
+- The selection of a single `StellarValue` from the set of candidates
+is the simplest way of ensuring that the combination function, given a
+set of self-consistent candidates, produces a consistent externalized value.
+
+Overall, changing the selection of `upgrades` to match that of `closeTime`
+could be said to make the resulting protocol simpler -- but the CAP change
+itself more complicated.  (We did start with and do some testing of an
+implementation that also changed `upgrades` selection, but in the latest
+implementation we have undone that part and are changing only `closeTime`.)
+
+Regarding potential performance impacts, using `SIGNED` `StellarValue`s in the
+ballot protocol will result in the ballot protocol checking signatures in some
+code paths where there were previously no signatures to check.  However, the
+signatures will be from nodes that had nominated values in the latest round of
+the nomination protocol, so the signature cache might ameliorate any noticeable
+effects of this.
+
+## Backwards Incompatibilities
+
+Semantic changes resulting from this CAP we hope to be limited to the
+correction of the potential inconsistency of smart contract behavior
+described in the "Motivation".  Smart contracts, and other clients, should
+not have _relied_ on that inconsistency, as it requires a race to occur,
+and one which is probably very rare.
+
+## Security Concerns
```

Comment:
This section is about the security concerns for this proposal. If there were security concerns that caused us not to choose other proposals, then that should be in the design rationale.

Reply:
Okay -- it was in the "Design Rationale" already, so I simply removed it as redundant from the "Security Concerns" section (and referenced the "Design Rationale" section there).